runtime/sema/checker.go

```diff
@@ -2119,8 +2119,8 @@ func (checker *Checker) accessFromAstAccess(access ast.Access) (result Access) {
 						)
 					}
 				}
-				result = PrimitiveAccess(ast.AccessNotSpecified)
-				return
+				semanticEntitlements = append(semanticEntitlements, NewEntitlementType(checker.memoryGauge, checker.Location, "<<INVALID>>"))
```

Comment:
I see how we need a `*EntitlementType` here, and `InvalidType` is not one.

The new `EntitlementType` constructed here does not indicate itself that it is an invalid type. Maybe  add a flag to `EntitlementType` to indicate if it is invalid, and return it from  `EntitlementType.IsInvalidType`.

Instead of the magic string it should then likely also be fine to use the identifier of the AST element (`entitlement`)